pkg/controller/command/command_controller.go

```diff
@@ -357,8 +374,7 @@ func (r *ReconcileCommand) Reconcile(request reconcile.Request) (reconcile.Resul
 		if err := r.prepareIntendedDeployment(deployment, command); err != nil {
 			return err
 		}
-		performUpgradeStepIfNeeded(command, deployment, oldDeploymentSpec)
-		return nil
+		return r.performUpgradeStepIfNeeded(command, deployment, oldDeploymentSpec, commandBootStrapConfigName)
```

Comment:
[Not for this PR] I don't really understand why this performUpgradeStepIfNeeded method is inside  this `CreateOrUpdate`. When I look at this I think it only makes things harder. Original deployment needs to be copied, in some states we create a new deployment and then it is overwritten by old configurations...

Reply:
unfortunately yes, i'm trying to figure out how to do it better way [not in this PR] :)